core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -123,7 +123,9 @@ public synchronized ClientStream newStream(
       return new NoopClientStream() {
         @Override
         public void start(ClientStreamListener listener) {
-          listener.closed(shutdownStatus, new Metadata());
+          synchronized (InProcessTransport.this) {
+            listener.closed(shutdownStatus, new Metadata());
```

Comment:
why is this needed? How would the listener be called asynchronously if we're only just starting it now?